File: frameworks/typescript-playwright/config/environment.ts
const requireValue = (name: 'API_KEY' | 'UI_EMAIL' | 'UI_PASSWORD'): string => {
  const value = process.env[name]?.trim();
  if (!value) throw new Error(`${name} is required. Copy .env.example to .env.`);
  return value;
};

export const environment = {
  baseUrl: (process.env.BASE_URL ?? 'https://qaacademyabc.xyz').replace(/\/$/, ''),
  get apiKey() {
    return requireValue('API_KEY');
  },
  get uiEmail() {
    return requireValue('UI_EMAIL');
  },
  get uiPassword() {
    return requireValue('UI_PASSWORD');
  },
};

export function vercelAutomationHeaders(): Record<string, string> {
  const secret = process.env.VERCEL_AUTOMATION_BYPASS_SECRET?.trim();
  if (!secret) return {};
  return {
    'x-vercel-protection-bypass': secret,
    'x-vercel-set-bypass-cookie': 'true',
  };
}

File: frameowkrs/java-selenium-rest-assured/config/Environment.java
package config;

public final class Environment {
    public static final String BASE_URL =
            System.getenv().getOrDefault("BASE_URL", "https://qaacademyabc.xyz").replaceAll("/$", "");

    private Environment() {}

    public static String required(String name) {
        String value = System.getenv(name);
        if (value == null || value.isBlank()) {
            throw new IllegalStateException(name + " is required. Copy .env.example to .env.");
        }
        return value.trim();
    }
}
